### DISCUSSION TITLE
Working with 3D Tiles in Pyvista

### DISCUSSION THREAD

**Discussion by Jimmy-KL** [pyvista/pyvista#3854] [Ideas]
When working with 3D geospatial data or models, there is a data format that we cannot avoid - 3D Tiles. If we are working on a large-scale model such as a geological or geophysics model, we have 2 ways to present it on the web efficiently - remote rendering or use 3D Tiles to stream the data on the client side.
We can process the data using Pyvista, then export the scene or the processed data to 3D tiles so that it can be loaded in something like CesiumJS. Here is another benefit - the vtk model can be presented together with other models such as the CityGML model, photographic model, point cloud, and the globe with maps.
There is a 3d tiles exporter in VTK for polydata mesh.  I haven't digged into it. I don't know whether it can export the scalars data together or only geometry. If there is only geometry, I think it's not very useful.
However, I think VTK is not the right tool to load 3D Tiles as it's not built to stream data on the client. Anyway, if VTK can, it would be better. But converting VTK model into 3D Tiles should have priority. If Pyvista can do the conversion, it would be great.
BTW， I found an article about this topic on Kitware's [web](https://www.kitware.com/3d-tiles-generation-using-vtk/).

**Comment by danlipsa** [comment]
> I don't know whether it can export the scalars data together or only geometry. If there is only geometry, I think it's not very useful

We haven't experienced with additional scalars besides the RGB colors for point clouds, so I am not sure if it would work out of the box. It would not be difficult to add though. All the code needed should be there.

Fig 5. in the blog you referenced, shows a 9 component scalar associated with faces (as a texture) rather than associated with points.

> However, I think VTK is not the right tool to load 3D Tiles as it's not built to stream data on the client.

Indeed, we used Cesium.js to show the 3D tiles data, just because it is already available. VTK / VTK.js would be better suited to do visualization, so if there is a need for that it would be an interested project to add 3D Tiles streaming to VTK.js
